addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -371,20 +302,182 @@ var SlideDialog = Widget.extend({
+                                page_content = page_content.replace(/\0/g, "");
+                                self.index_content = self.index_content + page_number + ". " + page_content + '\n';
+                                if (maxPages === page_number) {
+                                    if (loaded) {
+                                        self.$('.save').button('reset');
+                                    }
+                                    loaded = true;
+                                }
+                            });
+                        });
+                    }
+                });
+            };
+        }
+
+        var input = file.name;
+        var input_val = input.substr(0, input.lastIndexOf('.')) || input;
+        this._checkUniqueSlide(input_val).then(function (exist) {
+            if (exist) {
+                var message = _t("Channel contains the given title, please change before Save or Publish.");
+                self._displayAlert(message);
+            }
+            self.$('#name').val(input_val);
+        });
+    },
+    /**
+     * @override
+     * @param {Object} ev
+     */
+    _slideUrl: function (ev) {
+        var self = this,
+            value = {
+                'url': $(ev.target).val(),
+                'channel_id': self.channel_id
+            };
+        this.$('.alert-warning').remove();
+        this.is_valid_url = false;
+        this.$('.save').button('loading');
+        ajax.jsonRpc('/slides/dialog_preview/', 'call', value).then(function (data) {
+            self.$('.save').button('reset');
+            if (data.error) {
+                self._displayAlert(data.error);
+            } else {
+                self.$("#slide-image").attr("src", data.url_src);
+                self.$('#name').val(data.title);
+                self.$('#description').val(data.description);
+                self.is_valid_url = true;
+            }
+        });
+    },
 
-// bind the event to the button
-$('.oe_slide_js_upload').on('click', function () {
-    var channel_id = $(this).attr('channel_id');
-    slides.page_widgets['upload_dialog'] = new SlideDialog(this, channel_id).appendTo(document.body);
 });
 
-// automatically open the upload dialog if requested from query string
-if ($.deparam.querystring().enable_slide_upload !== undefined) {
-    $('.oe_slide_js_upload').click();
-}
+sAnimations.registry.websiteSlidesUpload = sAnimations.Class.extend({
+    selector: '.oe_slides_box_shadow',
+    read_events: {
+        'click .oe_slide_js_upload': '_bindEventButton',
+    },
+
+    /**
+     * @override
+     * @param {Object} parent
+     */
+    start: function (parent) {
+        console.log('start');
+        if (!$('.oe_slide_js_upload').length) {
+            return $.Deferred().reject("DOM doesn't contain '.oe_slide_js_upload'");
+        }
+
+        ajax.loadXML('/website_slides/static/src/xml/website_slides.xml', qweb);
```

Comment:
Another advantage of using animations: you can replace this by the use of `xmlDependencies`, check example in the code ;)